src/eventer/eventer_SSL_fd_opset.c

```diff
@@ -937,39 +942,88 @@ static int next_proto_cb(SSL *ssl, const unsigned char **data,
 static int alpn_select_proto_cb(SSL *ssl, const unsigned char **out,
                                 unsigned char *outlen, const unsigned char *in,
                                 unsigned int inlen, void *arg) {
-  int rv;
-  (void)ssl;
-  eventer_SSL_alpn_func_t f = (eventer_SSL_alpn_func_t)arg;
-
-  rv = f(out, outlen, in, inlen);
+  (void)arg;
+  if(!ssl) return SSL_TLSEXT_ERR_NOACK;
+  eventer_ssl_ctx_t *ctx = SSL_get_eventer_ssl_ctx(ssl);
+  if(!ctx) return SSL_TLSEXT_ERR_NOACK;
+  if(!ctx->npn) return SSL_TLSEXT_ERR_NOACK;
```

Comment:
Does this leak ``ctx``?

Reply:
No, ctx is a refernce there and not allocated.